inc/media.php

```diff
@@ -1058,11 +1058,18 @@ function media_preview_buttons($image, $auth, $rev='') {
  * @return array|false
  */
 function media_image_preview_size($image, $rev, $meta, $size = 500) {
-    if (!preg_match("/\.(jpe?g|gif|png)$/", $image) || !file_exists(mediaFN($image, $rev))) return false;
-
-    $info = getimagesize(mediaFN($image, $rev));
-    $w = (int) $info[0];
-    $h = (int) $info[1];
+    if (!file_exists(mediaFN($image, $rev))) return false;
+ 
+    if (preg_match("/\.svg$/", $image) ) {
+        $xml = simplexml_load_file(mediaFN($image, $rev));
+        $attr = $xml->attributes();
+        $w = (int) $attr->width ;
+        $h = (int) $attr->height ;
```

Comment:
Well, I assume most users will not configure the attributes themselves. The svgpureinsert plugin works around this problem by reconfiguring the SVGs to use 100% for width and height and setting an appropriate aspect ratio view box.
But yes, simply setting the a max-width and max-height and then including the SVG as is without specifying any dimensions would probably (mostly) work.